src/Lykke.Service.ClientAssetRule.Services/AssetConditionLayerRuleService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Common;
+using Common.Log;
+using Lykke.Service.Assets.Client;
+using Lykke.Service.ClientAssetRule.Core.Domain.AssetConditionLayer;
+using Lykke.Service.ClientAssetRule.Core.Repositories;
+using Lykke.Service.ClientAssetRule.Core.Services;
+using Lykke.Service.ClientAssetRule.Core.Utils;
+
+namespace Lykke.Service.ClientAssetRule.Services
+{
+    public class AssetConditionLayerRuleService : IAssetConditionLayerRuleService
+    {
+        private readonly IAssetConditionLayerRuleRepository _assetConditionLayerRuleRepository;
+        private readonly IAssetsService _assetsService;
+        private readonly ILog _log;
+
+        public AssetConditionLayerRuleService(
+            IAssetConditionLayerRuleRepository assetConditionLayerRuleRepository,
+            IAssetsService assetsService,
+            ILog log)
+        {
+            _assetConditionLayerRuleRepository = assetConditionLayerRuleRepository;
+            _assetsService = assetsService;
+            _log = log;
+        }
+
+        public async Task<IEnumerable<IAssetConditionLayerRule>> GetAsync()
+        {
+            return await _assetConditionLayerRuleRepository.GetAsync();
+        }
+
+        public async Task<IAssetConditionLayerRule> GetAsync(string regulationId)
+        {
+            return await _assetConditionLayerRuleRepository.GetAsync(regulationId);
+        }
+
+        public async Task SetAsync(string clientId, IEnumerable<string> regulations)
+        {
+            IList<IAssetConditionLayerRule> rules = await _assetConditionLayerRuleRepository.GetAsync();
+
+            List<string> allowed = rules
```

Comment:
HashSet would be better. https://github.com/LykkeCity/Lykke.Service.ClientAssetRule/pull/5/files#diff-44b0aadf07a12762f1652a019a22befaR54